#! /bin/sh /usr/share/dpatch/dpatch-run
## 14_fix_video_fallback.dpatch by Michael Haas <laga+mythtv@laga.ath.cx>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Don't fall back to opengl video output immediately if eg xvmc-blit 
## DP: is not available. xv-blit is better. 
## DP: see http://svn.mythtv.org/trac/ticket/4951

@DPATCH@

--- a/libs/libmythtv/videoout_xv.cpp      (revision 16772)
+++ b/libs/libmythtv/videoout_xv.cpp      (working copy)
@@ -4313,12 +4313,12 @@
     QStringList list;
     if (myth_codec_id < kCodec_NORMAL_END)
     {
-        if (opengl)
-            list += "opengl";
         if (xv)
             list += "xv-blit";
         if (shm)
             list += "xshm";
+        if (opengl)
+            list += "opengl";
         list += "xlib";
     }
     else

